File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        //has table to solve this one
        int size = nums.size();
        vector<int> arr;
        arr.resize(size + 1, 0);

        for(int i  = 0; i < nums.size(); i++) {
            arr[nums[i]] = 1;
        }

        vector<int> ans;
        for(int i = 1; i <= nums.size(); i++) {
            if(arr[i] == 0)
                ans.push_back(i);
        }

        return ans;
    }
};

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        #using set store the numbers which are there already
        #iterate and find the missing one
        set_nums = set(nums)
        missed_numbers = []

        for req_num in range(1, len(nums) + 1):
            if req_num not in set_nums:
                missed_numbers.append( req_num )
        
        return missed_numbers